src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableExtensions.cs

```diff
@@ -106,6 +106,10 @@ internal static int GetCount<T>(ref IEnumerable<T> sequence)
         /// </remarks>
         internal static bool TryCopyTo<T>(this IEnumerable<T> sequence, T[] array, int arrayIndex)
         {
+            Debug.Assert(sequence != null);
+            Debug.Assert(array != null);
+            Debug.Assert(arrayIndex >= 0 && arrayIndex < array.Length);
```

Comment:
> Also, it appears this method simply returned false when sequence was null before. Now it will throw NullReferenceException. Are we sure no callers pass a null 1st argument (perhaps as a result of an as cast that fails)?

@AArnott I double-checked and there are no callers doing that.

> Are we certain this last assert is correct? I seem to recall some APIs allowing arrayIndex == array.Length when the number of elements to be copied was 0.

I did that intentionally because I thought this method was only being called in scenarios where count > 0. However, I just realized I introduced a usage in this PR where count can be 0. I'll change this to `<=`.